opensafely/check.py

```diff
@@ -0,0 +1,115 @@
+from opensafely._vendor import requests
+from opensafely._vendor.ruamel.yaml import YAML
+import os
+import re
+import glob
+import subprocess
+
+DESCRIPTION = "Check the opensafely project for correctness"
+
+RESTRICTED_DATASETS = {"icnarc": ["admitted_to_icu"]}
+
+# todo: dummy location and data
+PERMISSIONS_URL = "https://raw.githubusercontent.com/Jongmassey/research_repository_permissions/main/repository_permissions.yaml"
+
+
+def add_arguments(parser):
+    pass
+
+
+def main():
+    repo_name = get_repository_name()
+    permissions = get_datasource_permissions(PERMISSIONS_URL)
+    allowed_datasets = get_allowed_datasets(repo_name, permissions)
+    datasets_to_check = {
+        k: v
+        for k, v in RESTRICTED_DATASETS.items()
+        if k not in allowed_datasets
+    }
+    files_to_check = get_local_py_files()
+
+    found_datasets = {
+        dataset: check_dataset(functions, files_to_check)
+        for dataset, functions in datasets_to_check.items()
+        if check_dataset(functions, files_to_check)
+    }
+
+    if found_datasets:
+        exit(print_violations(found_datasets))
```

Comment:
nit: python convention would fully qualify the exit.

It is a global function too, but that's really intended for interactive console use.

```python
import sys
...
sys.exit(...)
```